fastcrypto/src/groups/multiplier/integer_utils.rs

```diff
@@ -189,3 +193,17 @@ mod tests {
         assert!(is_power_of_2(4096));
     }
 }
+
+// We implementation `ToLittleEndianByteArray` for BigInt in case it needs to be used as scalar for
+// multi-scalar multiplication.
+impl<const N: usize> ToLittleEndianByteArray<N> for BigInt {
+    fn to_le_byte_array(&self) -> [u8; N] {
+        let mut output = [0u8; N];
+        let bytes = self.to_bytes_le().1;
+
+        // It's up to the caller to ensure that the BigInt is not too large to fit in N bytes.
```

Comment:
it's a bit scary, why not return Result<> in this case?

Reply:
True. It was to avoid error handling in msm. It's fixed by this [commit](https://github.com/MystenLabs/fastcrypto/pull/706/commits/999edf943137709281068cdccfb75c8e365837f6).